upstairs/src/lib.rs

```diff
@@ -1544,7 +1544,8 @@ impl Downstairs {
         &mut self,
         ds_id: u64,
         client_id: u8,
-        read_data: &Result<Vec<ReadResponse>, CrucibleError>,
+        responses: Result<Vec<ReadResponse>, CrucibleError>,
+        encryption_context: &Option<Arc<EncryptionContext>>,
```

Comment:
This method, like many, is growing in how much it does under the name of "complete".  I think if we add any more to it, then it might be time to break it up